### ISSUE TITLE
Ranking with EasyMKL/KOMD?

### ISSUE THREAD

**Issue by MaximilianLombardo** [jmikko/EasyMKL#2]
Hey Jmikko,

Thank you for the implementation. Do you have any tips/examples on how to use these algorithms for ranking?

Thanks,

Max

**Comment by jmikko** [comment]
Dear Max,

both the examples (toytest_komd.py and toytest_EasyMKL.py) solve a ranking task.

To be precise, concerning KOMD, you can use the method "decision_function" to obtain a real value for each test example (and consequentially an induced ranking).
In the example of usage, the line 56:
y_score = classifier.fit(X[idtrain,:], Ytr).decision_function(X[idtest,:])

where X[idtrain,:] is the training examples matrix, Ytr are the training labels and X[idtest,:] is the test examples matrix. It is important to note that - in this line of code - you are performing the training (by using the fit method) and then the prediction (by using the decision_function method).

Concerning EasyMKL, the method is called "rank". In the example of usage, the line 64:
ranktest = np.array(easy.rank(klistte))

Best,
Michele.

P.s.
EasyMKL and KOMD are in the Python package MKLpy:
https://pypi.python.org/pypi/MKLpy
You can install it by using the command: pip install MKLpy

**Comment by MaximilianLombardo** [comment]
Hi Michele!

I've gotten a bit sidetracked over the past few days (hence the delayed reply). I appreciate your response and understand how to use the package for ranking I think. I will close the issue once I get everything working if that is ok with you ;)

One question, it seems like there are different methods for the EasyMKL class in this repository's version vs the MKLpy version. Other than the names of the methods, these two classes should be the same though right?

Best

Max

**Comment by jmikko** [comment]
Yes, concerning EasyMKL, they are very similar.

MKLpy is a package scikit-compliant that contains more than only EasyMKL.

**Comment by MaximilianLombardo** [comment]
Hey Michele,

For the ranking example included here, this is actually ranking different kernels correct? It is not showing how to approach a traditional ranking problem with EasyMKL, right?

Also reading the paper, you guys are using EasyMKL for a combination of many weak kernels, but do you think EasyMKL could be used for a few strong kernels which have been derived from orthogonal data views? I would like to use the resulting combined kernel for a regression/ranking problem.

best,

max

**Comment by jmikko** [comment]
You have, of course, a ranking of the kernels (the weights).
BUT EasyMKL provides also a classifier. This classifier returns a real value for each example, and consequentially you have also a ranking of the examples (from the most "positive" to the most "negative" ones)

Concerning your second question, EasyMKL can handle a huge number of weak kernel. On the other hand, as a standard MKL algorithm, it is also able to handle a small familiy of strong orthogonal kernels.

best,
Michele.

**Comment by MaximilianLombardo** [comment]
Regarding your first point: This could also be use for regression then? Is it possible to plug the summed kernel into an SVR machine from SKlearn for instance? My targets are real valued, won't this affect training?

**Comment by MaximilianLombardo** [comment]
Hey Michele,

I just saw your comment to the other issue, so I guess that regression is indeed not possible with EasyMKL at the moment

**Comment by jmikko** [comment]
You are right.
At the moment is not possibile, due to the fact that KOMD (the model behind EasyMKL) is a binary classifier.

By the way, I think an extension is possible and not very difficult.

**Comment by MaximilianLombardo** [comment]
Ok thanks
